ui/app/AppLayouts/Chat/views/MembersSelectorView.qml

```diff
@@ -121,8 +114,18 @@ MembersSelectorBase {
 
             if (root.model.count === 0 && !hasPendingContactRequest) {
                 // List is empty and not a contact yet. Open the contact request popup
-                Global.openContactRequestPopup(contactDetails.publicKey,
-                                               popup => popup.closed.connect(root.rejected))
+
+                // If `displayName` is not undefiend and not empty,
+                // then we open the popup with given `contactData`, which probably came from URL.
+                if (contactData.displayName) {
```

Comment:
what if you have a data without public key?

Reply:
Is that possible? 😮

Reply:
We can't send a contact request then 😄

Reply:
At the first time, I thought that link will be processed first (but probably, should be checked isContact for it). Otherwise, data which was embedded to the link will be useless. If its okay - then okay :)

Reply:
The logic seems to be correct here.

1. `getContactDetailsAsJson`
- this's only to check local information about contact
- we don't request anything from waku and don't drop given `contactData`.
2. check if `publicKey` is not empty 
(this could probably be done before (1), but anyway
3. check `isContact`
- not possible without (1)
4. check `hasPendingContact
- not possible without (1)
5. Finally, open contact request modal
- with `contactData`, if `displayName` is not empty and not undefined
- else without `contactData` (which will make a request from waku on popup opening

This looks fine to me, wdyt?

Reply:
yeah it looks correct :)